CombineX.podspec

```diff
@@ -27,19 +27,17 @@ Pod::Spec.new do |s|
     ss.source_files = "Sources/CXNamespace/**/*.swift"
   end
 
-  s.subspec "Main" do |ss|
-    ss.source_files = "Sources/CombineX/**/*.swift"
-    ss.dependency "CombineX/CXLibc"
-    ss.dependency "CombineX/CXUtility"
-    ss.dependency "CombineX/CXNamespace"
-    # ss.dependency "Runtime"
-  end
-
   s.subspec "CXFoundation" do |ss|
     ss.source_files = "Sources/CXFoundation/**/*.swift"
     ss.dependency "CombineX/CXUtility"
     ss.dependency "CombineX/CXNamespace"
-    ss.dependency "CombineX/Main"
+  end
+
+  s.subspec "Main" do |ss|
+    ss.source_files = "Sources/CombineX/**/*.swift"
+    ss.dependency "CombineX/CXLibc"
+    ss.dependency "CombineX/CXFoundation"
```

Comment:
CXFoundation is built upon Main, not the opposite. See [Package.swift](https://github.com/cx-org/CombineX/blob/master/Package%40swift-5.2.swift) for project structure.

Users do need to specify CXFoundation subspec explicitly. Changing this behavior is out of scope of this pr.